packages/compiler/core/parser.ts

```diff
@@ -611,24 +613,44 @@ export function parse(code: string | SourceFile, options: ParseOptions = {}): Ca
   function parseOperationStatement(
     pos: number,
     decorators: DecoratorExpressionNode[]
-  ): OperationStatementNode {
+  ): OperationStatementNode | OperationInstanceNode {
     parseExpected(Token.OpKeyword);
 
     const id = parseIdentifier();
-    const parameters = parseOperationParameters();
-    parseExpected(Token.Colon);
+    const templateParameters = parseTemplateParameterList();
 
-    const returnType = parseExpression();
-    parseExpected(Token.Semicolon);
```

Comment:
I think you'll get better parser errors if you use `parseExpectedOneOf(Token.OpenParen, Token.Colon)` here.

Reply:
Looks like I can't do that because I'm using `parseOperationParameters` which expects to see the `OpenParen` so I can't consume it beforehand.